### DISCUSSION TITLE
Could not read the "SUPPORTED_PLATFORMS" build setting, assuming that the project supports iOS only. && Hang before checking out swiftPM.

### DISCUSSION THREAD

**Discussion by megabitsenmzq** [fastlane/fastlane#22108] [Q&A]
I'm using fastlane with GitHub Action self-hosted runner. The lane runs flawlessly before when using BitBucket self-hosted runner.

I have two problems:

The first problem is that "build_app" hangs while "resolved source packages". After downloading all the packages, it hangs before checking out.

```
INFO [2024-06-24 12:19:21.65]: ▸ Command line invocation:
INFO [2024-06-24 12:19:21.65]: ▸     /Applications/Xcode-16.0.0-Beta.app/Contents/Developer/usr/bin/xcodebuild -resolvePackageDependencies -scheme "MenubarX Paddle" -project ./MenubarX.xcodeproj -configuration Release
INFO [2024-06-24 12:19:21.65]: ▸ User defaults from command line:
INFO [2024-06-24 12:19:21.65]: ▸     IDEPackageSupportUseBuiltinSCM = YES
INFO [2024-06-24 12:19:21.84]: ▸ Resolve Package Graph
INFO [2024-06-24 12:19:22.24]: ▸ Fetching from https://github.com/megabitsenmzq/FaviconFinder (cached)
INFO [2024-06-24 12:19:23.26]: ▸ Fetching from https://github.com/megabitsenmzq/MuteableWKWebView (cached)
INFO [2024-06-24 12:19:24.08]: ▸ Fetching from https://github.com/mixpanel/mixpanel-swift (cached)
INFO [2024-06-24 12:19:24.73]: ▸ Fetching from https://github.com/apple/swift-log (cached)
INFO [2024-06-24 12:19:24.73]: ▸ Fetching from https://github.com/sparkle-project/Sparkle (cached)
INFO [2024-06-24 12:19:24.73]: ▸ Fetching from https://github.com/rollbar/rollbar-apple (cached)
INFO [2024-06-24 12:19:25.35]: ▸ Fetching from https://github.com/apple/swift-docc-plugin (cached)
INFO [2024-06-24 12:19:25.35]: ▸ Fetching from https://github.com/MacPaw/Setapp-framework.git (cached)
INFO [2024-06-24 12:19:25.35]: ▸ Fetching from https://github.com/CocoaLumberjack/CocoaLumberjack.git (cached)
INFO [2024-06-24 12:19:26.08]: ▸ Fetching from https://github.com/dagronf/DSFAppearanceManager (cached)
INFO [2024-06-24 12:19:26.08]: ▸ Fetching from https://github.com/dagronf/DSFPagerControl (cached)
INFO [2024-06-24 12:19:26.08]: ▸ Fetching from https://github.com/apple/swift-docc-symbolkit (cached)
INFO [2024-06-24 12:19:26.80]: ▸ Fetching from https://github.com/scinfu/SwiftSoup.git (cached)
```

⬆️ It stops here.

I try to use "xcodebuild -resolvePackageDependencies" before fastlane and disable "dependencies resolution" in fastlane. It passed this step.

Then the second problem happened.
```
WARN [2024-06-24 12:23:45.05]: Could not read the "SUPPORTED_PLATFORMS" build setting, assuming that the project supports iOS only.
WARN [2024-06-24 12:23:45.05]: Could not read the "SUPPORTED_PLATFORMS" build setting, assuming that the project supports iOS only.
WARN [2024-06-24 12:23:45.05]: Could not read the "SUPPORTED_PLATFORMS" build setting, assuming that the project supports iOS only.
WARN [2024-06-24 12:23:45.05]: Could not read the "SUPPORTED_PLATFORMS" build setting, assuming that the project supports iOS only.
```

This caused my build to fall because the target is a macOS app.

I searched for these problems all day and still have no resolution. Do you guys have any ideas?

**Comment by DesmanLead** [comment]
@megabitsenmzq have you managed to set the shell locale to UTF-8?
We're trying to solve a similar issue and noticed that self-hosted runner uses US-ASCII by default. Not sure what's the right place to change it as the runner ignores `.bash_profile`.

UPD:
Done it using the `.env` file at the runner root.

**Reply by DesmanLead** [reply]
Setting the encoding fixed the hang in our case.
Here's how the `.env` should look like:
```
LC_ALL=en_US.UTF-8
LANG=en_US.UTF-8
```
Also, had to add the `unlock_keychain` step for `codesign` to succeed. Now it works 🎉

**Reply by megabitsenmzq** [reply]
Thank you for helping me. I already gave up on self hosting my CI, so I can't test it for now. I might give it a try someday.

**Reply by JqyModi** [reply]
Thanks, Successfully solved the problem

**Comment by gsabran** [comment]
Did you figure out how to resolve the `Could not read the "SUPPORTED_PLATFORMS" build setting, assuming that the project supports iOS only.` issue? [It's happening for me](https://github.com/getcmd-dev/cmd/actions/runs/18297797040/job/52099636628) on a regular action runner (not self hosted)
